From: Clemens Gruber <clemens.gruber@pqgruber.com>
Date: Tue, 6 Jun 2017 16:10:51 +0200
Subject: [PATCH] Fix scanner config for flex 2.6.x

Signed-off-by: Clemens Gruber <clemens.gruber@pqgruber.com>
---
 src/config-flex.l | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/config-flex.l b/src/config-flex.l
index 433992f26c34..dd59614da6d3 100644
--- a/src/config-flex.l
+++ b/src/config-flex.l
@@ -108,15 +108,13 @@ OPTIONS				{ BEGIN(options); return OPTIONS; }
 
 %%
 
-#ifndef yywrap
-int yywrap()
+int lt_config_wrap()
 {
 	return 1;
 	/* XXX not to get the compiler 'not used' warning */
 	yyunput(0, NULL);
 	input();
 }
-#endif
 
 void lt_config_error(const char *m)
 {
