From: Antonio Alvarez Feijoo <antonio.feijoo@suse.com>
Date: Thu, 26 Dec 2024 11:02:34 +0100
Subject: [PATCH] vmspawn-register: add missing header for glibc < 2.34

Fixes #35755
---
 src/vmspawn/vmspawn-register.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/vmspawn/vmspawn-register.c b/src/vmspawn/vmspawn-register.c
index 20e6def360fb..e60467496adb 100644
--- a/src/vmspawn/vmspawn-register.c
+++ b/src/vmspawn/vmspawn-register.c
@@ -1,5 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+#include <unistd.h>
+
 #include "sd-bus.h"
 #include "sd-id128.h"
 #include "sd-json.h"
