From: Jean-Marc Valin <jmvalin@jmvalin.ca>
Date: Tue, 12 Mar 2024 02:03:22 -0400
Subject: [PATCH] Fix meson AVX2 fixed-point

---
 silk/meson.build | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/silk/meson.build b/silk/meson.build
index 80a59b050eb5..35d955784f5d 100644
--- a/silk/meson.build
+++ b/silk/meson.build
@@ -44,9 +44,7 @@ foreach intr_name : ['sse4_1', 'avx2', 'neon_intr']
   endif
 
   intr_sources = get_variable('silk_sources_' + intr_name)
-  if opt_fixed_point
-    intr_sources += get_variable('silk_sources_fixed_' + intr_name)
-  else
+  if not opt_fixed_point
     intr_sources += get_variable('silk_sources_float_' + intr_name)
   endif
 
