From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 3 Dec 2020 15:29:01 +0100
Subject: [PATCH] meson: fix building with -Dbuild_introspection_data=false

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 14af2b17109b..8429553d483e 100644
--- a/meson.build
+++ b/meson.build
@@ -224,7 +224,7 @@ subdir('docs')
 
 # The tests will also run, which is not possible if they
 # were built for a different architecture.
-if not meson.is_cross_build()
+if not meson.is_cross_build() and get_option('build_introspection_data')
     subdir('tests')
 endif
 
@@ -283,7 +283,7 @@ pkg.generate(
 # FIXME: meson.override_dependency() and declare_dependency()'s variable arguments
 # are new in Meson 0.54.0, older versions of Meson won't be able to use g-i as
 # subproject anyway
-if meson.version().version_compare('>=0.54.0')
+if meson.version().version_compare('>=0.54.0') and get_option('build_introspection_data')
   girepo_dep = declare_dependency(
     sources: typelibs,
     dependencies: girepo_dep,
