From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 14 Aug 2020 09:44:34 +0200
Subject: [PATCH] SConstruct: don't make python files without shebang
 executable

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 SConstruct | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 434141f3e46b..d303c3aa6d20 100644
--- a/SConstruct
+++ b/SConstruct
@@ -2116,7 +2116,7 @@ for fn in templated:
         env.Default(builder)
     # set read-only to alert people trying to edit the files.
     env.AddPostAction(builder, 'chmod -w $TARGET')
-    if ((fn.endswith(".py.in") or
+    if ((fn[:-3] in ['contrib/skyview2svg.py', 'gps/gps.py'] or
          fn[:-3] in python_progs or
          fn[:-3] in ['contrib/ntpshmviz', 'contrib/webgps'])):
         # set python files to executable
