From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 3 Apr 2020 13:07:30 +0200
Subject: [PATCH] don't try ru build / run code generator

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 lib/nettle/Makefile.am | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/nettle/Makefile.am b/lib/nettle/Makefile.am
index aae87e09023a..de6f9ed44399 100644
--- a/lib/nettle/Makefile.am
+++ b/lib/nettle/Makefile.am
@@ -99,6 +99,8 @@ ecc_generated_headers = \
 BUILT_SOURCES = $(ecc_generated_headers)
 EXTRA_DIST = $(ecc_generated_headers) ecc/eccdata.stamp
 
+if !CROSS_COMPILING
+
 noinst_PROGRAMS = ecc/eccdata$(EXEEXT)
 
 ecc_eccdata_SOURCES = ecc/eccdata.c
@@ -126,6 +128,8 @@ ecc/ecc-gost-gc512a-32.h: ecc/eccdata.stamp
 ecc/ecc-gost-gc512a-64.h: ecc/eccdata.stamp
 	$(AM_V_GEN)ecc/eccdata$(EXEEXT) gost_gc512a 43 6 64 > $@T && mv $@T $@
 
+endif
+
 libcrypto_la_SOURCES += \
 	gost/ecc-gost-hash.c nettle-alloca.h ecc-gost-curve.h
 
