From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 4 Sep 2023 15:11:51 +0200
Subject: [PATCH] HACK: use existing meson without venv

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 configure | 53 +----------------------------------------------------
 1 file changed, 1 insertion(+), 52 deletions(-)

diff --git a/configure b/configure
index 133f4e323512..6f0d45a4dcc3 100755
--- a/configure
+++ b/configure
@@ -1009,58 +1009,7 @@ python="$(command -v "$python")"
 echo "python determined to be '$python'"
 echo "python version: $($python --version)"
 
-python="$($python -B "${source_path}/python/scripts/mkvenv.py" create pyvenv)"
-if test "$?" -ne 0 ; then
-    error_exit "python venv creation failed"
-fi
-
-# Suppress writing compiled files
-python="$python -B"
-mkvenv="$python ${source_path}/python/scripts/mkvenv.py"
-
-mkvenv_flags=""
-if test "$download" = "enabled" ; then
-    mkvenv_flags="--online"
-fi
-
-if ! $mkvenv ensure \
-     $mkvenv_flags \
-     --dir "${source_path}/python/wheels" \
-     --diagnose "meson" \
-     "meson>=0.63.0" ;
-then
-    exit 1
-fi
-
-# At this point, we expect Meson to be installed and available.
-# We expect mkvenv or pip to have created pyvenv/bin/meson for us.
-# We ignore PATH completely here: we want to use the venv's Meson
-# *exclusively*.
-
-meson="$(cd pyvenv/bin; pwd)/meson"
-
-# Conditionally ensure Sphinx is installed.
-
-mkvenv_flags=""
-if test "$download" = "enabled" -a "$docs" = "enabled" ; then
-    mkvenv_flags="--online"
-fi
-
-if test "$docs" != "disabled" ; then
-    if ! $mkvenv ensure \
-         $mkvenv_flags \
-         --diagnose "sphinx-build" \
-         "sphinx>=1.6.0" "sphinx-rtd-theme>=0.5.0";
-    then
-        if test "$docs" = "enabled" ; then
-            exit 1
-        fi
-        echo "Sphinx not found/usable, disabling docs."
-        docs=disabled
-    else
-        docs=enabled
-    fi
-fi
+meson="meson"
 
 # Probe for ninja
 
