From: unknown author <unknown.author@example.com>
Date: Mon, 24 May 2021 22:44:25 +0200
Subject: [PATCH] build reproducibly

Origin: Debian, https://sources.debian.org/patches/minicom/2.8-1/04reproducible.diff/
---
 src/minicom.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/minicom.c b/src/minicom.c
index c9ce75c223be..8a006a3e039f 100644
--- a/src/minicom.c
+++ b/src/minicom.c
@@ -1323,9 +1323,6 @@ int main(int argc, char **argv)
       switch(c) {
         case 'v':
           printf(_("%s version %s"), PACKAGE, VERSION);
-#ifdef __DATE__
-          printf(_(" (compiled %s)"), __DATE__);
-#endif
           printf("\n");
           printf(_("Copyright (C) Miquel van Smoorenburg.\n\n"));
           printf(_("This program is free software; you can redistribute it and/or\n"
@@ -1659,9 +1656,6 @@ int main(int argc, char **argv)
 
   mc_wprintf(us, "\n%s %s\r\n", _("Welcome to minicom"), VERSION);
   mc_wprintf(us, "\n%s: %s\r\n", _("OPTIONS"), option_string);
-#if defined (__DATE__) && defined (__TIME__)
-  mc_wprintf(us, "%s %s, %s.\r\n",_("Compiled on"), __DATE__,__TIME__);
-#endif
   {
     struct stat st;
     char port_date[20] = "";
