From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sat, 18 Apr 2020 10:52:34 +0200
Subject: [PATCH] configure: handle gcc 9.x

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index a8e6dc141eda..39c895b091f3 100755
--- a/configure
+++ b/configure
@@ -7734,7 +7734,7 @@ case "$XPLATFORM" in
     *-g++*)
 	# Check gcc's version
 	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
-	    8*|7*|6*|5*|4*|3.4*)
+	    9*|8*|7*|6*|5*|4*|3.4*)
 		;;
             3.3*)
                 canBuildWebKit="no"
