From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Tue, 24 Nov 2015 17:11:37 +0100
Subject: [PATCH] Makefile: don't create absolute link outside of DESTDIR

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 599b1452a05a..0a2bacacfb88 100644
--- a/Makefile
+++ b/Makefile
@@ -200,7 +200,7 @@ ifeq ($(NO_SOLIB),0)
 	$(INSTALL) -D $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	$(LNS) $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	mkdir -p $(DESTDIR)$(USRLIBDIR)
-	$(LNS) $(LIBDIR)/$(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
+	$(LNS) $(SONAME) $(DESTDIR)$(LIBDIR)/$(DEVELLIB)
 	sed \
 	-e 's,@VERSION\@,$(VERSION),g' \
 	-e 's,@prefix\@,$(PREFIX),g' \
