From 0e7a0cafda6599d4af114fc0154a90b8786af7c6 Mon Sep 17 00:00:00 2001
From: Artur Wiebe <artur@4wiebe.de>
Date: Thu, 13 Apr 2023 16:47:50 +0200
Subject: [PATCH] add setup.py

---
 setup.py | 6 ++++++
 1 file changed, 6 insertions(+)
 create mode 100644 setup.py

diff --git a/setup.py b/setup.py
new file mode 100644
index 0000000..358320c
--- /dev/null
+++ b/setup.py
@@ -0,0 +1,6 @@
+from setuptools import setup
+
+setup(
+	name='aiosqlite',
+	packages=['aiosqlite'],
+)
-- 
2.40.0

