From: Clemens Gruber <clemens.gruber@pqgruber.com>
Date: Mon, 15 Nov 2021 16:21:31 +0100
Subject: [PATCH] Support systemd and set a data directory

---
 redis.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/redis.conf b/redis.conf
index 97f077b0d0aa..abc03980de3e 100644
--- a/redis.conf
+++ b/redis.conf
@@ -324,7 +324,7 @@ daemonize no
 # The default is "no". To run under upstart/systemd, you can simply uncomment
 # the line below:
 #
-# supervised auto
+supervised auto
 
 # If a pid file is specified, Redis writes it where specified at startup
 # and removes it at exit.
@@ -507,7 +507,7 @@ rdb-del-sync-files no
 # The Append Only File will also be created inside this directory.
 #
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir /var/lib/redis/
 
 ################################# REPLICATION #################################
 
