From: Federico Fuga <fuga@studiofuga.com>
Date: Thu, 23 May 2019 13:40:39 +0200
Subject: [PATCH] Fix #39 Compilation fails on gcc v8

---
 src/usbg_common.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/usbg_common.c b/src/usbg_common.c
index f8822fcf727f..f3aa8b053c44 100644
--- a/src/usbg_common.c
+++ b/src/usbg_common.c
@@ -20,6 +20,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <sys/sysmacros.h>
 
 int usbg_read_buf_limited(const char *path, const char *name,
 			  const char *file, char *buf, int len)
