From: "Gary E. Miller" <gem@rellim.com>
Date: Wed, 8 Jan 2020 14:56:58 -0800
Subject: [PATCH] SConstruct: Remove redundant CheckHeaderDefines()

---
 SConstruct | 19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

diff --git a/SConstruct b/SConstruct
index d71dfab48db3..1a0463f67a41 100644
--- a/SConstruct
+++ b/SConstruct
@@ -581,22 +581,6 @@ def CheckCompilerOption(context, option):
     return ret
 
 
-def CheckHeaderDefines(context, file, define):
-    context.Message('Checking if %s supplies %s... ' % (file, define))
-    ret = context.TryLink("""
-        #include <%s>
-        #ifndef %s
-        #error %s is not defined
-        #endif
-        int main(int argc, char **argv) {
-            (void) argc; (void) argv;
-            return 0;
-        }
-    """ % (file, define, define), '.c')
-    context.Result(ret)
-    return ret
-
-
 def CheckCompilerDefines(context, define):
     context.Message('Checking if compiler supplies %s... ' % (define,))
     ret = context.TryLink("""
@@ -670,7 +654,6 @@ config = Configure(env, custom_tests={
     'CheckC11': CheckC11,
     'CheckCompilerDefines': CheckCompilerDefines,
     'CheckCompilerOption': CheckCompilerOption,
-    'CheckHeaderDefines': CheckHeaderDefines,
     'CheckPKG': CheckPKG,
     'CheckXsltproc': CheckXsltproc,
     'GetPythonValue': GetPythonValue,
@@ -1046,7 +1029,7 @@ else:
         if config.env["magic_hat"]:
             announce("Forcing magic_hat=no since RFC2783 API is unavailable")
             config.env["magic_hat"] = False
-    tiocmiwait = config.CheckHeaderDefines("sys/ioctl.h", "TIOCMIWAIT")
+    tiocmiwait = config.CheckDeclaration("TIOCMIWAIT", "#include <sys/ioctl.h>")
     if not tiocmiwait and not kpps:
         announce("Neither TIOCMIWAIT nor RFC2783 API is available)")
         if config.env["timeservice"]:
